llama-index-core/llama_index/core/workflow/workflow.py

```diff
@@ -328,7 +333,8 @@ def _validate(self) -> None:
 
                 produced_events.add(event_type)
 
-            requested_services.update(step_config.services.keys())
+            for sd in step_config.requested_services:
```

Comment:
ooc: I guess we couldn't just use an `update()` here with `step_config.requested_services`?

Reply:
Nope, it's a leftover! Changing to `update` rn